boards/native/native_posix/board.yml

```diff
@@ -0,0 +1,9 @@
+boards:
+- name: native_posix
+  vendor: Zephyr
+  socs:
+  - name: native
+- name: native_posix_64
```

Comment:
should it be the board or the soc that is considered 64bit ?

Meaning, should we instead have:
```
board:
  name: native_posix
  vendor: Zephyr
  socs:
  - name: native
  - name: native_64
```

Considering the SoC 64bit  then that would also remove the need for:
```
config BOARD_NATIVE_POSIX_64
	 bool
	 select BOARD_NATIVE_POSIX
	 select 64BIT
```
as the 64bit is then selected by the soc and there is no need to `select BOARD_NATIVE_POSIX`.
I also think this aligns more with what is done for `qemu` boards.

@nordicjm thoughts ?

Reply:
> should it be the board or the soc that is considered 64bit ?
I intended to do that as a follow up. In this PR I hoped to bring everything back to working order in the hwmv2 format while keeping the PR reasonably sized. The next steps were to define this and the nrf5340 as having different socs , change all tests and BT test infra accordingly, and then remove this "backwards compatible" board definitions.

@nordicjm 's pointed issue would still apply for the nrf*bsim boards, there the nrf52 will remain separate from the nrf53 anyhow.

Reply:
true, and it's a side-effect of allowing multiple boards being defined in board.yml.

Something which would needs to be addressed.